[yuvrajaryan] IP

yuvrajaryan opened the pull request:
Simon

> “Your mind is for having ideas, not holding them.” – David Allen ([[source](https://dansilvestre.com/productivity-quotes)])


Simon frees your mind of having to remember things you need to do. It's,


- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,

1. download it from [[here](https://github.com/yuvrajaryan/ip/releases/tag/A-Jar)].
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
5. And it is **FREE**!


Features:

- [x]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(Simon.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

yuvrajaryan committed:
Added .class to gitignore and renamed the chatbot to simon

yuvrajaryan committed:
Added echo functionality

yuvrajaryan committed:
Added echo functionality

yuvrajaryan committed:
Merge branch 'week-2-changes'

yuvrajaryan committed:
Added add and list functionality

yuvrajaryan committed:
Added add and list functionality

yuvrajaryan committed:
Merge branch 'week-2-changes'

yuvrajaryan committed:
Added mark as done functionality

yuvrajaryan committed:
Merge branch 'week-2-changes'

yuvrajaryan committed:
Added ToDos, Events, Deadlines as done functionality

yuvrajaryan committed:
Merge branch 'week-2-changes'

yuvrajaryan committed:
Automated Text UI-Testing

yuvrajaryan committed:
Merge branch 'week-2-changes'

yuvrajaryan committed:
Added error handling

yuvrajaryan committed:
Merge branch 'week-2-changes'

yuvrajaryan committed:
Added delete function

yuvrajaryan committed:
Merge branch 'week-2-changes'

yuvrajaryan committed:
Adds save function

yuvrajaryan committed:
saves deadline in datetime format

yuvrajaryan committed:
improve OOP

yuvrajaryan committed:
adds all files to a package

yuvrajaryan committed:
Merge branch 'add-gradle-support'

yuvrajaryan committed:
automates build process with gradle

yuvrajaryan committed:
resolves merge conflict in build.gradle

yuvrajaryan committed:
Adds JUnit tests

yuvrajaryan committed:
Merge branch 'branch-A-More'

yuvrajaryan committed:
Adds JAR

yuvrajaryan committed:
Adds JAR

yuvrajaryan committed:
Add JavaDoc comments

yuvrajaryan committed:
Fixes coding standard

yuvrajaryan committed:
Implement find functionality

yuvrajaryan committed:
Resolve conflicts in branch-A-CodingStandard merge

yuvrajaryan committed:
Merge branch 'branch-Level-9'

yuvrajaryan committed:
aligns code with the stylechecker

yuvrajaryan committed:
adds a functional GUI.

QinHaichen12 commented on src/main/java/simon/Deadline.java:
```diff
@@ -0,0 +1,60 @@
+ * Inherits from the Task class and adds a deadline property.
+ */
+
+public class Deadline extends Task {
```
Maybe rename it to DeadlineTask?

QinHaichen12 commented on src/main/java/simon/Task.java:
```diff
@@ -0,0 +1,101 @@
+     *
+     * @return {@code true} if the task is completed, {@code false} otherwise.
+     */
+    public Boolean getCompleted() {
```
Maybe name boolean methods as isComplete or hasCompleted?

QinHaichen12 commented on src/main/java/simon/TaskList.java:
```diff
@@ -0,0 +1,107 @@
+     * @param mark If {@code true}, the task will be marked as done; otherwise, it will be marked as not done.
+     * @param index The index of the task to be marked.
+     */
+    public void markTask(boolean mark, int index) {
```
rename mark to isMarked?

QinHaichen12 commented on src/main/java/simon/TaskList.java:
```diff
@@ -0,0 +1,107 @@
+     *
+     * @return An ArrayList containing all tasks in the task list.
+     */
+    public ArrayList<Task> toArr() {
```
Maybe rename to getTasks?

QinHaichen12 reviewed (commented):
LGTM! Very well structured and adheres to coding style. Good job buddy!

slidings commented on src/main/java/simon/Task.java:
```diff
@@ -0,0 +1,101 @@
+     *
+     * @return the task number
+     */
+    public int getNumber() {
```
Valiant effort with the javadocs, could possibly forgo it for simple methods like setters and getters?

slidings commented on src/main/java/simon/Command.java:
```diff
@@ -0,0 +1,15 @@
+     * @param ui the user interface to interact with the user
+     * @param storage the storage to save or retrieve data
+     */
+    public String execute(TaskList taskList, Ui ui, Storage storage);
```
javadoc for overriden methods could perhaps be left out to concrete methods that implements it?

slidings reviewed (commented):
Overall code looks really good to me 🔥 🔥 
Hard to even find mistakes to nitpick about. Great job! 👍 👍 

Edit: Small nitpick - There are some small formatting improvements that could be made by not having blank line between javadoc and method. (I think it's mostly just careless)

yuvrajaryan committed:
Adds assertions

yuvrajaryan committed:
Add more assertions

TaskList: add assertions to ensure task integrity

Task operations do not currently enforce certain assumptions, which could
lead to unintended behavior if these assumptions are violated.

Asserts help catch potential issues during development by validating these
assumptions at runtime.

Let's,
* add assertions to validate that tasks are not null before adding
* add assertions to ensure the command is not null before processing

yuvrajaryan committed:
Code refactoring

Parser: refactor command parsing to improve readability and maintainability

The `parse` method uses a long series of `else if` statements to determine
which command to execute based on the user's input, which reduces readability
and increases the complexity of adding new commands.

A more modular approach would make the code easier to understand, maintain,
and extend.

Let's,
* refactor the `parse` method by extracting each command's logic into
  separate helper methods
* introduce a command map to store commands and their parsing logic,
  reducing code duplication and improving maintainability

yuvrajaryan committed:
Improve code quality

yuvrajaryan committed:
Merge pull request #3 from yuvrajaryan/branch-A-CodeQuality

Branch a code quality

yuvrajaryan committed:
Add duplicate task check in TaskList

Updated the TaskList class to include duplicate task detection in the add method. The changes ensure that a task is only added if it does not already exist in the list.

- Implemented a check to prevent adding tasks with duplicate string representations.
- Used the task's string representation to identify duplicates.
- Added assertions to ensure the task being added is not null.

The add method now performs the following steps:
1. Asserts that the task is not null.
2. Checks if the task (by its string representation) already exists in the list.
3. If not a duplicate, the task is added to the list.
4. If the task was marked as done during the check, it is reverted to not done before adding.

This update improves the integrity of the task list by avoiding duplicate entries.

yuvrajaryan committed:
Merge branch 'branch-A-CodeQuality'

yuvrajaryan committed:
Adds CI/CD workflow

yuvrajaryan committed:
Fix style problems in code

yuvrajaryan committed:
Create .gitlab-ci.yml

yuvrajaryan committed:
Updated parser test

yuvrajaryan committed:
Merge branch 'master' of https://github.com/yuvrajaryan/ip

yuvrajaryan committed:
Changed CI/CD location

yuvrajaryan committed:
Fixed style issues

yuvrajaryan committed:
update .yml file

yuvrajaryan committed:
Rename boolean to sound more boolean-like

yuvrajaryan committed:
Rename booleans to sound more boolean-like

yuvrajaryan committed:
Remove dead code from task.java

yuvrajaryan committed:
Fix javadocs for main function in Simon.java

yuvrajaryan committed:
Remove instances of duke and update the header of gui

yuvrajaryan committed:
Fix style errors

yuvrajaryan committed:
Add ui screenshot

yuvrajaryan committed:
Rename ui.png to Ui.png

yuvrajaryan committed:
Update README.md

yuvrajaryan committed:
Merge branch 'master' into patch-1

yuvrajaryan committed:
Merge pull request #5 from yuvrajaryan/patch-1

Update README.md

yuvrajaryan committed:
Update README.md

yuvrajaryan committed:
Update README.md

yuvrajaryan committed:
Update README.md

yuvrajaryan committed:
Update README.md

yuvrajaryan committed:
Update README.md

yuvrajaryan committed:
Update README.md

yuvrajaryan committed:
Update README.md
